MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,72 @@
+﻿using AutoMapper;
+using MeetupPlatformApi.Authentification;
+using MeetupPlatformApi.Context;
+using MeetupPlatformApi.DataTransferObjects;
+using MeetupPlatformApi.Entities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Claims;
+
+namespace MeetupPlatformApi.Controllers
+{
+    [Route("/api/meetups/authentication")]
+    [ApiController]
+    public class AuthenticationController : ControllerBase
+    {
+        private readonly ApplicationContext context;
+        private readonly IMapper mapper;
+        private readonly AuthentificationManager authentificationManager;
+
+        public AuthenticationController(ApplicationContext context, IMapper mapper, AuthentificationManager authentificationManager)
+        {
+            this.context = context;
+            this.mapper = mapper;
+            this.authentificationManager = authentificationManager;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetUserById(Guid id)
+        {
+            var user = await context.Users.SingleOrDefaultAsync(u => u.Id.Equals(id));
+            var userOutput = mapper.Map<UserOutputDto>(user);
+
+            return userOutput == null ? Ok(userOutput) : NotFound();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> RegisterUser([FromBody] UserForCreationDto userFromBody)
+        {
+            var user = mapper.Map<UserEntity>(userFromBody);
+
+            user.Password = BCrypt.Net.BCrypt.HashPassword(user.Password);
+
+            await context.Users.AddAsync(user);
+            await context.SaveChangesAsync();
+
+            var userOutput = mapper.Map<UserOutputDto>(user);
+
+            return CreatedAtAction(nameof(GetUserById), new { id = userOutput.Id }, userOutput);
+        }
+
+        [HttpPost("login")]
```

Comment:
Authentication actions aren't precisely mapping to the common URL path conventions, but I'm absolutely sure that we can find a better option than `/api/meetups/authentication/login` (especially considering that "login" term is usually applied to the session authentications and we are using JWTs instead of sessions).

I've found a fine (not perfect, but still looks reasonable) resource name – `tokens`:
- for authentication we can use `POST /api/tokens` endpoint (kind of creating token);
- for refresh we can use `PUT /api/tokens` endpoint (kind of updating tokens).
This approach have following issues:
- when we start using Refresh Tokens, we will be posting token pair, so we might want to switch to `token-pairs` later on;
- refresh action doesn't have resource identifier;
- refresh request is not idempotent.

I'd appreciate if You suggested something more suiting, but if we have no other options, I'd strongly recommend switching to the `tokens` approach.